miden-tx/src/testing/tx_context.rs

```diff
@@ -43,22 +46,85 @@ use crate::{DataStore, DataStoreError};
 pub struct TransactionContext {
     mock_chain: MockChain,
     expected_output_notes: Vec<Note>,
-    tx_args: TransactionArgs,
     tx_inputs: TransactionInputs,
     advice_inputs: AdviceInputs,
 }
 
+/// Data required to compile the [miden_objects::transaction::TransactionScript] under the [TransactionContext].
+///
+/// This allows the account code to be loaded in the [Assembler] so that the transaction script can
+/// the account's procedures.
+pub enum ScriptAndInputs {
+    Empty,
+    Some {
+        source: String,
+        inputs: Vec<(Word, Vec<Felt>)>,
+        targets: Vec<ScriptTarget>,
+    },
+}
```

Comment:
This is a bit annoying, the tx script doesn't allow phantom calls, so the script must be compiled with an assembler which had the account loaded into it.

As the tests are currently setup, the assembler is kinda "hidden" inside the context. This structure is just to allow the test to pass script code that calls the account, and to have it compiled with the same assembler. Which then finally allows the tests to run against `main.masm` instead of stand alone scripts, which is leading me to find issues like the one this PR is fixing.